activerecord/lib/active_record/relation/spawn_methods.rb

```diff
@@ -37,7 +38,7 @@ def merge(other)
     end
 
     def merge!(other) # :nodoc:
-      if !other.is_a?(Relation) && other.respond_to?(:to_proc)
+      if other.is_a?(Proc)
```

Comment:
Sure pixeltrix's implementation supports `to_proc` duck typing, but I am not sure we need support duck typing here.

Docs of `merge` only says

> Procs will be evaluated by merge:

so we need support Procs can be passed at least.